core/src/reserved.rs

```diff
@@ -88,6 +88,14 @@ impl ReservedState {
                 tx.data.delegator
             ));
         }
+        if let Some(key) = self.query_public_key(&tx.data.delegator) {
+            if &key != tx.proof.signer() {
+                return Err("sign is not matched".to_string());
```

Comment:
```suggestion
                return Err("the key used for the proof does not match the key in the reserved state".to_string());
```
1. I think using simple present tense for error messages is better
2. I think it is better to have a descriptive error message